### PR TITLE
Improving logging from `AbstractCloudSlave._terminate`

### PR DESCRIPTION
**Pull request by jglick**
Since inception (f62fff5c0ca5091e9e7bc5945ea3a582ee3db3a4), `AbstractCloudSlave._terminate` implementations were supplied with a `TaskListener` that just streams to stdout of the JVM, something we should never do. (For example, it will not be captured by `support-core`.)

### Proposed changelog entries

* Improve logging for termination of cloud agents.

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/hudson/slaves/SlaveComputer.java
Comment (jglick) [inline core/src/main/java/hudson/slaves/SlaveComputer.java:221]: https://github.com/jenkinsci/jenkins/pull/2395#discussion_r66434831

**Review by basil — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by res0nance — APPROVED** [review]

**Pull request merged**
